packages/components/src/custom-gradient-picker/index.tsx

```diff
@@ -82,11 +98,13 @@ const GradientTypePicker = ( { gradientAST, hasGradient, onChange } ) => {
 		);
 	};
 
-	const handleOnChange = ( next ) => {
-		if ( next === 'linear-gradient' ) {
+	const handleOnChange = ( next: string | string[] ) => {
+		// NTS: Because `SelectControl.onChange` (below) can accept an array, we need to accept one here. We shouldn't ever see an array here, but I'm adding a check just in case.
+		const nextValue = Array.isArray( next ) ? next[ 0 ] : next;
```

Comment:
Because `SelectControl.onChange` (below) can accept an array, we need to accept one here. I don't think we should ever actually see an array here, but I'm adding a check just in case.

Reply:
Very nice, those changes made this much cleaner. Updated in cf33a070b4